### ISSUE TITLE
Can't return an adaptive card in response to a composeExtension/query

### ISSUE THREAD

**Issue by chrisbardon** [OfficeDev/BotBuilder-MicrosoftTeams-dotnet#15]
I'm trying to return an adaptive card as a response to composeExtension/query, and the teams client fails to display it ("something went wrong with this app").  This is using .net core 2.1, microsoft.bot.builder.teams v 4.0.0-beta1, and Microsoft.bot.builder 4.2.2.  Here's the response I'm trying to construct (note, this is using the adaptive card source from the sample):

            MessagingExtensionResult result = new MessagingExtensionResult
            {
                Type = "result",
                AttachmentLayout = "list",
                Attachments = new List<MessagingExtensionAttachment>(),
            };
            AdaptiveCard card = new AdaptiveCard("1.0");

            card.Body.Add(new AdaptiveTextBlock()
            {
                Text = "Hello",
                Size = AdaptiveTextSize.ExtraLarge
            });

            card.Body.Add(new AdaptiveImage()
            {
                Url = new Uri("http://adaptivecards.io/content/cats/1.png")
            });

            result.Attachments.Add(new Attachment() { ContentType = AdaptiveCard.ContentType, Content = card }.ToMessagingExtensionAttachment());

**Comment by RamjotSingh** [comment]
Please use Microsoft.bot.builder 4.2.1. There is a known issue with latest Bot.Builder package https://github.com/OfficeDev/BotBuilder-MicrosoftTeams-dotnet/issues/13

**Comment by chrisbardon** [comment]
I tried reverting Microsoft.bot.builder to 4.2.0 (4.2.1 isn't on nuget anymore), and I'm still seeing the same issue.  I also needed to revert microsoft.bot.builder.integration.aspnet.core, Microsoft.botconfiguration, Microsoft.bot.connector, and microsoft.bot.schema to 4.2.0 to get the app to run.

**Comment by RamjotSingh** [comment]
Good to know you are unblocked :). We are working with Bot.Builder for a fix for this issue. Closing this since the issue is being tracked in another issue already.

**Comment by chrisbardon** [comment]
Actually, the issue is still there, so I'm not unblocked. Trying to return an adaptive card, as in the original bug, fails the same way on v 4.2.0.is there anything else I can try?

**Comment by RamjotSingh** [comment]
Can you check what response is being sent out? Whether the response is a 200 with payload or is your service throwing exception

**Comment by chrisbardon** [comment]
No exception-it's a 200 with payload (code is above for generating the card).  Otherwise I'm just creating an InvokeResponse with a 200 status.  If I sub out the Adaptive Card for a HeroCard it works, but with the adaptive card code above, nothing displays in Teams.

**Comment by siduppal** [comment]
@chrisbardon can you capture and share the JSON payload being returned by your bot? E.g. using Fiddler.

**Comment by chrisbardon** [comment]
I just tried serializing the response body, that the code in the original issue generates this:

{
	"composeExtension": {
		"attachmentLayout": "list",
		"type": "result",
		"attachments": [{
				"preview": {
					"contentType": "application/vnd.microsoft.card.adaptive",
					"contentUrl": null,
					"content": {
						"type": "AdaptiveCard",
						"version": "1.0",
						"body": [{
								"type": "TextBlock",
								"size": "extraLarge",
								"text": "Hello"
							}, {
								"type": "Image",
								"url": "http://adaptivecards.io/content/cats/1.png"
							}
						]
					},
					"name": null,
					"thumbnailUrl": null
				},
				"contentType": "application/vnd.microsoft.card.adaptive",
				"contentUrl": null,
				"content": {
					"type": "AdaptiveCard",
					"version": "1.0",
					"body": [{
							"type": "TextBlock",
							"size": "extraLarge",
							"text": "Hello"
						}, {
							"type": "Image",
							"url": "http://adaptivecards.io/content/cats/1.png"
						}
					]
				},
				"name": null,
				"thumbnailUrl": null
			}, {
				"preview": {
					"contentType": "application/vnd.microsoft.card.adaptive",
					"contentUrl": null,
					"content": {
						"type": "AdaptiveCard",
						"version": "1.0",
						"body": [{
								"type": "TextBlock",
								"text": "Test adaptive card"
							}
						]
					},
					"name": null,
					"thumbnailUrl": null
				},
				"contentType": "application/vnd.microsoft.card.adaptive",
				"contentUrl": null,
				"content": {
					"type": "AdaptiveCard",
					"version": "1.0",
					"body": [{
							"type": "TextBlock",
							"text": "Test adaptive card"
						}
					]
				},
				"name": null,
				"thumbnailUrl": null
			}
		],
		"suggestedActions": null,
		"text": null
	}
}

**Comment by RamjotSingh** [comment]
@siduppal Can you check the payload above?

**Comment by chrisbardon** [comment]
I checked the payload against adaptivecards.io, and taking a snippet like this:

 {
     "type": "AdaptiveCard",
     "version": "1.0",
     "body": [{
     "type": "TextBlock",
     "text": "Test adaptive card"
     }
     ]    
 }
It comes up as valid, so I believe what's there should work.

**Comment by chrisbardon** [comment]
Any updates on this issue?

**Comment by robin-liao** [comment]
@chrisbardon  - `preivew` can be only either Hero or Thumbnail cards. Sorry about that the online document doesn't clearly state it. Please check out constructor of [MessagingExtensionAttachment](https://github.com/OfficeDev/BotBuilder-MicrosoftTeams-dotnet/blob/487f9654df319b6426f5e41fd59eb3bf570fc7a6/CSharp/Microsoft.Bot.Schema.Teams/Generated/MessagingExtensionAttachment.cs#L40) to assign `preview` attachments.

**Comment by mayureshjaiswal** [comment]
> @chrisbardon - `preivew` can be only either Hero or Thumbnail cards. Sorry about that the online document doesn't clearly state it. Please check out constructor of [MessagingExtensionAttachment](https://github.com/OfficeDev/BotBuilder-MicrosoftTeams-dotnet/blob/487f9654df319b6426f5e41fd59eb3bf570fc7a6/CSharp/Microsoft.Bot.Schema.Teams/Generated/MessagingExtensionAttachment.cs#L40) to assign `preview` attachments.

Hi Robit, 
I want to send adaptive card in messaging extension, and as per above comments its not possible so can you please provide a supporting reference for the same.

**Comment by Javatar96** [comment]
Hi guys,
I kinda have the same issue with the search-based Message Extension. 
When I debug locally, my app works fine. after deploying the messaging extension on Azure and testing it in the Teams app, I can search for specific queries. The Message Extension gives me the desired results, up to here we are good to go.
The Problem:
After I select one of the delivered search results, the messaging extension does not return the adaptive card. The chat-box in Teams is simply empty. 

Is anyone familiar with this issue?
